src/Illuminate/Database/Eloquent/Relations/Concerns/AsPivot.php

```diff
@@ -110,11 +114,22 @@ protected function setKeysForSaveQuery(Builder $query)
      */
     public function delete()
     {
+        // support for pivot classes that container a non-composite primary key
         if (isset($this->attributes[$this->getKeyName()])) {
-            return parent::delete();
+            return (int) parent::delete();
```

Comment:
🤔

Reply:
`Model::delete()` returns either a boolean or an integer. `false` is taken to mean nothing got updated. The return value of this particular method is always an integer, thus casting.